### PR TITLE
introducing cosets Fixing #7127

### PR DESCRIPTION
**Pull request by kaushik94**
Currently it is not possible to simply G_a or a_G in sympy

### CODE REVIEW

**Comment by pelegm** [comment]
From travis:
```pytb
AssertionError: File contains trailing whitespace: /home/travis/virtualenv/python2.6.9/lib/python2.6/site-packages/sympy-0.7.6_git-py2.6.egg/sympy/combinatorics/perm_groups.py, line 222.
```

**Comment by kaushik94** [comment]
@pelegm  fixed !

**Comment by kaushik94** [comment]
@asmeurer  please review.

File: sympy/combinatorics/perm_groups.py [hunk 1]
```diff
@@ -217,9 +217,11 @@ def __mul__(self, other):
 
         """
         gens1 = [perm._array_form for perm in self.generators]
+        if isinstance(other, Permutation) or isinstance(other, Cycle):
+            return [i*other for i in [perm._array_form for perm in self.generators]]
```
Comment (urkud) [inline sympy/combinatorics/perm_groups.py]: Why not just [perm._array_form*other for perm in self.generators]?

File: sympy/combinatorics/perm_groups.py [hunk 2]
```diff
@@ -217,9 +217,11 @@ def __mul__(self, other):
...
         gens2 = [perm._array_form for perm in other.generators]
-        n1 = self._degree
         n2 = other._degree
+        n1 = self._degree
```
Comment (urkud) [inline sympy/combinatorics/perm_groups.py]: Any reason to reorder assignments?

Comment (kaushik94) [inline sympy/combinatorics/perm_groups.py]: No, I forgot I had something else in mind. Will change that.

File: sympy/combinatorics/perm_groups.py (hunk 1 above)
Comment (kaushik94) [inline sympy/combinatorics/perm_groups.py]: Right !. Thanks :)

**Comment by kaushik94** [comment]
Fixes #7127

**Pull request closed without merging**